addons/website/static/src/builder/plugins/form/utils.js

```diff
@@ -156,6 +156,24 @@ export function renderField(field, resetId = false) {
     template.content.querySelectorAll("[data-name]").forEach((el) => {
         el.dataset.name = getQuotesEncodedName(el.dataset.name);
     });
+    // Preserve “Other” option dataset on the re-rendered field.
+    if (["selection", "many2one"].includes(field.type)) {
+        const previousEl = field.formatInfo.fieldEl.querySelector(
+            ".form-select, .s_website_form_multiple"
+        );
+        const currentEl = template.content.querySelector(".form-select, .s_website_form_multiple");
+
+        if (previousEl && currentEl && previousEl.dataset.otherOptionAllowed) {
+            const { otherOptionAllowed, otherOptionLabel, otherOptionPlaceholder } =
+                previousEl.dataset;
```

Comment:
In case you keep this logic, a few improvements are possible.
```suggestion
        if (previousEl?.dataset.otherOptionAllowed && currentEl) {
            const { otherOptionAllowed, otherOptionLabel = "Other", otherOptionPlaceholder = "" } =
                previousEl.dataset;
```
You can set default values during destructuring so you don't need separate fallback checks later.